### PR TITLE
Net block: use nl80211 to get WiFi information

### PR DESCRIPTION
**Pull request by MaxVerevkin**
Net block: use https://github.com/Eonm/nl80211 to get WiFi information.

### CODE REVIEW

File: src/blocks/net.rs [hunk 1]
```diff
@@ -709,159 +668,8 @@ struct IpAddrInfo {
...
-    if let Some(raw_ssid) = raw {
-        String::from_utf8(decode_escaped_unicode(raw_ssid))
-            .block_error("net", "Non-UTF8 SSID")
-            .map(Some)
-    } else {
-        Ok(None)
-    }
-}
-
+// TODO use it?
```
Comment (ammgws) [inline src/blocks/net.rs:671]: If not used then will it be a regression of the issue https://github.com/greshake/i3status-rust/pull/995 solved?

**Comment by MaxVerevkin** [comment]
Did `cargo-clippy` update and now we are getting a ton of new warnings?

**Comment by ammgws** [comment]
I just updated from 

clippy 0.0.212 (cb75ad5 2021-02-10)
rustc 1.50.0 (cb75ad5db 2021-02-10)

to 

clippy 0.1.51 (2fd73fa 2021-03-23)
rustc 1.51.0 (2fd73fabe 2021-03-23)

and went from 0 warnings to 8, so looks like it yes.

**Comment by ammgws** [comment]
Hopefully one day we can get all the other parts of the net block going through netlink too.

Thanks! Looks much better now.

**Pull request merged**